### PR TITLE
Core Javadocs

### PR DESCRIPTION
**Pull request by SizableShrimp**
**Description:** This pull request is the start of documenting classes in core which do not have documentation. This is the first iteration and is open to revision and adding more documentation to other classes.

**Justification:** Add documentation so users know what methods and classes do in core.

### CODE REVIEW

File: core/src/main/java/discord4j/core/spec/AuditSpec.java [hunk 1]
```diff
@@ -18,10 +18,22 @@
 
 import reactor.util.annotation.Nullable;
 
+/** A spec which can optionally have a reason in the audit logs when built. */
 public interface AuditSpec<T> extends Spec<T> {
 
+    /**
+     * Sets the audit log reason to show in the audit logs when the spec is built.
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/AuditSpec.java]: I would word it "Sets the reason to show in the audit logs when the spec is built."

File: core/src/main/java/discord4j/core/spec/AuditSpec.java [hunk 2]
```diff
@@ -18,10 +18,22 @@
...
+     */
     AuditSpec<T> setReason(@Nullable String reason);
 
+    /**
+     * Returns the current audit log reason set on the spec.
+     *
+     * @return The current audit log reason.
+     */
     @Nullable
     String getReason();
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/AuditSpec.java]: This function like `Spec#asRequest` is not a truly supported method. Users should never call this, and that should be made clear.

File: core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java [hunk 3]
```diff
@@ -26,56 +26,125 @@
...
+ * A spec used to configure and send an embed.
+ * @see <a href="https://i.stack.imgur.com/HRWHk.png">Embed Layout</a>
+ */
 public class EmbedCreateSpec implements Spec<EmbedRequest> {
 
     private final EmbedRequest.Builder requestBuilder = EmbedRequest.builder();
     private final List<EmbedFieldEntity> fields = new ArrayList<>();
 
+    /**
+     * Sets the main title of the embed.
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java]: I think "main" is extraneous.

File: core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java [hunk 4]
```diff
@@ -26,56 +26,125 @@
...
+     * @param title The title of the embed.
+     * @return This spec.
+     */
     public EmbedCreateSpec setTitle(String title) {
         requestBuilder.title(title);
         return this;
     }
 
+    /**
+     * Sets the main description of the embed.
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java]: I think "main" is extraneous.

File: core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java [hunk 5]
```diff
@@ -26,56 +26,125 @@
...
+     */
     public EmbedCreateSpec setDescription(String description) {
         requestBuilder.description(description);
         return this;
     }
 
+    /**
+     * Sets the URL of the embed.
+     *
+     * @param url A URL which can be clicked on through the main title of the embed.
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/EmbedCreateSpec.java]: I think "main" is extraneous.

File: core/src/main/java/discord4j/core/spec/GuildCreateSpec.java [hunk 6]
```diff
@@ -29,6 +29,19 @@
 import java.util.List;
 import java.util.function.Consumer;
 
+/**
+ * A spec used to configure and create a {@link Guild}.
+ * <b>This can only be used for bots in less than 10 guilds.</b>
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/GuildCreateSpec.java]: The newlines in this Javadoc are weird. Keep it consistent and just keep going until you hit the endline limit.

File: core/src/main/java/discord4j/core/spec/VoiceChannelCreateSpec.java [hunk 7]
```diff
@@ -18,41 +18,81 @@
...
+     */
     public VoiceChannelCreateSpec setName(String name) {
         requestBuilder.name(name);
         return this;
     }
 
+    /**
+     * Sets the bitrate for the created {@link VoiceChannel}.
+     *
+     * @param bitrate The maximum amount of bits to send per second in the voice channel, related to the quality of audio. A valid bitrate is a number from 8 to 96.
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/VoiceChannelCreateSpec.java]: Is this going past the endline limit?

File: core/src/main/java/discord4j/core/spec/VoiceChannelCreateSpec.java [hunk 8]
```diff
@@ -18,41 +18,81 @@
...
     }
 
+    /**
+     * Sets the permission overwrites for the created {@link VoiceChannel}.
+     *
+     * @param permissionOverwrites The {@code Set<PermissionOverwrite>} which contains overwrites for the channel.
+     * @return This spec.
+     */
     public VoiceChannelCreateSpec setPermissionOverwrites(Set<? extends PermissionOverwrite> permissionOverwrites) {
+        Map<Character, Integer> myMap = new LinkedHashMap<>();
```
Comment (danthonywalker) [inline core/src/main/java/discord4j/core/spec/VoiceChannelCreateSpec.java]: Why is there code added here?

**Comment by quanticc** [comment]
@SizableShrimp

**Comment by SizableShrimp** [comment]
I will work on it soon; I have been busy with school and life recently.

File: core/src/main/java/discord4j/core/spec/AuditSpec.java (hunk 2 above)
Comment (SizableShrimp) [reply core/src/main/java/discord4j/core/spec/AuditSpec.java]: how do you want me to make it clear to users? aren't there annotations I can add to the method now? or I could notify them in the documentation or remove the docs for the method altogether.

**Comment by quanticc** [comment]
#489

**Pull request merged**